### PR TITLE
Nimble timeline

### PR DESCRIPTION
**Pull request by anuja01**
Timeline chart

### CODE REVIEW

File: src/components/nimbleTimeline/InfoLabel.tsx
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/InfoLabel.tsx:1]: Styled components moved to new `styleWrapper.tsx`

File: src/components/nimbleTimeline/InfoLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  color: 'white',
+  padding: 10,
+  fontSize: 20,
+  borderRadius: 5,
+  zIndex: 85,
+});
+export default ({group, time}: {group: any; time: any}) => {
+  const date = moment(time, 'x');
+  const label = group ? group.title : '';
+  return <LabelContainer style={{}}>{`${date.format('LLL')}, ${label} `}</LabelContainer>;
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/InfoLabel.tsx:27]: This was kept by mistake, and removed

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+const GroupRightSection = styled('div')({
+  display: 'flex',
+  height: 'inherit',
+  padding: '4px 0px',
+  width: '50%',
+  overflow: 'scroll',
+  alignItems: 'center',
+});
+
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx:87]: done

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+  itemDoubleClickHandler,
+}) => {
+  const [groups] = useState(sidebarGroups);
+  const [items, setItems] = useState(timelineItems);
+  const [draggedItem, setDraggedItem] = useState<{item: any; group: Group; time: number} | undefined>(undefined);
+
+  const defaultTimeStart = moment().startOf('day').toDate();
+  const defaultTimeEnd = moment().startOf('day').add(1, 'day').toDate();
+
+  var keys = {
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx:132]: yep, changed it

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+      itemDoubleClickHandler && itemDoubleClickHandler(item);
+    }
+  };
+
+  const handleItemMove = (itemId: number, dragTime: number, newGroupOrder: number) => {
+    const group = groups[newGroupOrder];
+    setItems(
+      items.map(item =>
+        item.id === itemId
+          ? Object.assign({}, item, {
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx]: Done

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+      ),
+    );
+    setDraggedItem(undefined);
+    itemMoveHandler && itemMoveHandler(itemId, dragTime, newGroupOrder);
+  };
+
+  const handleItemResize = (itemId: number, time: number, edge: string) => {
+    setItems(
+      items.map(item =>
+        item.id === itemId
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx:116]: Done

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+      <GroupContainer>
+        <GroupLeftSection>
+          {group?.badge && (
+            <Chip
+              label={group?.badge}
+              size={'small'}
+              sx={{
+                fontWeight: 100,
+                borderRadius: '3px',
+                margin: '0px 8px',
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx]: New styled component created

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+        <GroupRightSection>
+          {group?.labels?.map((label, index) => (
+            <Chip
+              key={index}
+              label={label.text}
+              size={'small'}
+              sx={{
+                fontWeight: 100,
+                borderRadius: '3px',
+                margin: '0px 4px',
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx]: New styled component created

File: src/components/nimbleTimeline/NimbleTimeline.tsx [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+} from 'react-calendar-timeline';
+
+import InfoLabel from './InfoLabel';
+import 'react-calendar-timeline/lib/Timeline.css';
+import './styles.css';
+
+interface NimbleTimeline {
+  showWeeks?: boolean;
+  sidebarWidth?: number;
+  sidebarGroups?: any[];
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/NimbleTimeline.tsx:32]: Groups and Items are passed to the Timeline component from the library, and things are getting messy when trying to define a specific type here.

File: src/components/nimbleTimeline/styles.css [hunk 9]
```diff
@@ -0,0 +1,64 @@
+@import url('https://fonts.googleapis.com/css?family=Work+Sans');
+
+body {
+    font-family: 'Work Sans', sans-serif;
```
[the remark this answers is not in the document]
Comment (anuja01) [reply src/components/nimbleTimeline/styles.css]: font-family is now a prop

**Review by Manoj201 — APPROVED** [review]

**Pull request merged**